shell/platform/android/io/flutter/plugin/localization/LocalizationPlugin.java

```diff
@@ -19,11 +22,56 @@
   @NonNull private final LocalizationChannel localizationChannel;
   @NonNull private final Context context;
 
+  @VisibleForTesting
+  final LocalizationChannel.LocalizationMessageHandler localizationMessageHandler =
+      new LocalizationChannel.LocalizationMessageHandler() {
+        @Override
+        public String getStringResource(@NonNull String key, @Nullable String localeString) {
+          Context localContext = context;
+          String stringToReturn = null;
+          Locale savedLocale = null;
+
+          if (localeString != null) {
+            Locale locale = localeFromString(localeString);
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+              Configuration conf = new Configuration(context.getResources().getConfiguration());
+              conf.setLocale(locale);
+              localContext = context.createConfigurationContext(conf);
+            } else {
+              // setLocale and createConfigurationContext is only available on API >= 17
+              Resources resources = context.getResources();
+              Configuration conf = resources.getConfiguration();
+              savedLocale = conf.locale;
+              conf.locale = locale;
+              resources.updateConfiguration(conf, null);
+            }
+          }
+
+          String packageName = context.getPackageName();
+          int resId = localContext.getResources().getIdentifier(key, "string", packageName);
+          if (resId != 0) {
+            // 0 means the resource is not found.
+            stringToReturn = localContext.getResources().getString(resId);
+          }
+
+          if (localeString != null && Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR1) {
```

Comment:
nit, comment the API number for `Build.VERSION_CODES.JELLY_BEAN_MR1`

Reply:
Done